tests/test_mig_cgibin_cat.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import print_function
+import importlib
+import os
+import shutil
+import sys
+
+from tests.support import MIG_BASE, MigTestCase, testmain, \
+    fixturefile, fixturefile_normname, \
+    _ensuredirs, _temppath
+
+from mig.shared.base import client_id_dir
+from mig.shared.functionality.cat import _main as main
```

Comment:
AFAICT what you are really unit testing here is `mig/shared/functionality/cat.py` and not `mig/cgi-bin/cat.py`.
Mind you, that is perfectly fine so, but I think that should be reflected both in the test filename and the Class name below. The references to wsgi_variables also appear slightly misleading in that context.
In short the functionality cat backend doesn't care which provider serves the output_objects to the user: wsgi, cgi or the xmlrpc/jsonrpc handlers on top for that matter.

This new test file is missing the mandatory copyright header, btw. Perhaps because you copy/pasted from `test_support.py`, where it was also missing until I added it a few minutes ago. Anyway, please add it and let me know if you want to give it another go with the above or if it is easier that I rename during merge.

Reply:
Yep, you're right - this test morphed a bit as it was first extracted from elsewhere then adjusted/reworked. With it's pre-requisites merged it is now rather simple for me to do another pass: rename the file and do some of these other adjustments. Think it right I do that as the author/submitter :)

Reply:
Great, thanks